ext/reflection/php_reflection.c

```diff
@@ -2954,26 +2954,41 @@ ZEND_METHOD(reflection_method, __construct)
 	size_t name_len, tmp_len;
 	zval ztmp;
 
-	if (zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS(), "zs", &classname, &name_str, &name_len) == FAILURE) {
-		if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "s", &name_str, &name_len) == FAILURE) {
-			return;
-		}
+	switch (ZEND_NUM_ARGS()) {
+		case 2:
+			if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "zs", &classname, &name_str, &name_len) == FAILURE) {
+				return;
+			}
 
-		if ((tmp = strstr(name_str, "::")) == NULL) {
-			zend_throw_exception_ex(reflection_exception_ptr, 0,
-				"Invalid method name %s", name_str);
-			return;
-		}
-		classname = &ztmp;
-		tmp_len = tmp - name_str;
-		ZVAL_STRINGL(classname, name_str, tmp_len);
-		name_len = name_len - (tmp_len + 2);
-		name_str = tmp + 2;
-		orig_obj = NULL;
-	} else if (Z_TYPE_P(classname) == IS_OBJECT) {
-		orig_obj = classname;
-	} else {
-		orig_obj = NULL;
+			if (Z_TYPE_P(classname) == IS_OBJECT) {
+				orig_obj = classname;
+			} else {
+				orig_obj = NULL;
+			}
+
+			break;
+
+		case 1:
+			if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "s", &name_str, &name_len) == FAILURE) {
+				return;
+			}
+
+			if ((tmp = strstr(name_str, "::")) == NULL) {
+				zend_throw_exception_ex(reflection_exception_ptr, 0, "Invalid method name %s", name_str);
+				return;
+			}
+
+			classname = &ztmp;
+			tmp_len = tmp - name_str;
+			ZVAL_STRINGL(classname, name_str, tmp_len);
+			name_len = name_len - (tmp_len + 2);
+			name_str = tmp + 2;
+			orig_obj = NULL;
+
+			break;
+
+		default:
+			ZEND_WRONG_PARAM_COUNT();
```

Comment:
```suggestion
			zend_wrong_parameters_count_exception(1, 2);
```
Should do the trick, for PHP 7.3 and 7.4, the person doing the merge-up to master/PHP 8.0 will need to change it to ``zend_wrong_parameters_count_error()`` as the former API has been removed in master.